packages/engine/lib/error/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+// TODO inheritence vs composition
```

Comment:
(It doesn't need to be done in this PR, but for reference, the nested enums we make with `thiserror` are heirarchical like inheritance, whereas this library is like composition.)

Reply:
Just noticed, that we already have something like this: https://github.com/hashintel/hash/blob/f0d38edf26eed368b14b932ed06ec607a317b0e5/packages/engine/lib/error/src/lib.rs#L23-L25

I could elaborate on this after the itemization